src/main/scala/no/ndla/draftapi/service/StateTransitionRules.scala

```diff
@@ -61,8 +69,14 @@ trait StateTransitionRules {
         article.id match {
           case Some(id) =>
             val externalIds = draftRepository.getExternalIdsFromId(id)
-            taxonomyApiClient.updateTaxonomyIfExists(id, article)
-            articleApiClient.updateArticle(id, article, externalIds, isImported, useSoftValidation)
+
+            val taxonomyTries = Seq(
+              taxonomyApiClient.updateTaxonomyIfExists(id, article),
+              taxonomyApiClient.updateTaxonomyMetadataIfExists(id, true)
+            )
+            val articleUdpT = articleApiClient.updateArticle(id, article, externalIds, isImported, useSoftValidation)
+            val failures = (taxonomyTries :+ articleUdpT).collect { case Failure(ex) => Failure(ex) }
+            failures.headOption.getOrElse(articleUdpT)
```

Comment:
Syns kanskje ikke dette er den peneste koden i verden, men jeg tror vi har lyst til å gjøre alle `Try`'ene selvom en av de feiler, men alltid returnere feil om noen feiler(?) så derfor passer det ikke så bra med en finere for-comprehension.

Kom gjerne med forslag til forbedring :)

Reply:
Du treng ikkje oppdatere visbleflagget ved publisering så denne koden kan fjernes..

Reply:
Aight, men skal vi ikke ha noen mulighet for å få `visible: true` igjen?

Reply:
Kun via strukturredigering, og det held Elena på med.

Reply:
:+1: